Sources/NIOHTTP1/HTTPDecoder.swift

```diff
@@ -561,7 +592,23 @@ public final class HTTPDecoder<In, Out>: ByteToMessageDecoder, HTTPDecoderDelega
                                           headers: HTTPHeaders(self.headers,
                                                                keepAliveState: keepAliveState))
             message = NIOAny(HTTPServerRequestPart.head(reqHead))
+            
+        case .response where (100..<200).contains(statusCode) && statusCode != 101:
+            self.lastHeaderWasInformal = true
+            
+            switch self.informalResponseStrategy {
+            case .forward:
+                let resHead: HTTPResponseHead = HTTPResponseHead(version: .init(major: versionMajor, minor: versionMinor),
+                                                                 status: .init(statusCode: statusCode),
+                                                                 headers: HTTPHeaders(self.headers,
+                                                                                      keepAliveState: keepAliveState))
+                message = NIOAny(HTTPClientResponsePart.head(resHead))
```

Comment:
Can I propose we wrap the common code in this case and the case below into a single helper?